backend/functions/healthcheck/healthcheck.db.cosmos.ts

```diff
@@ -31,57 +45,61 @@ export default class HealthcheckCosmosDb {
 
   public async checkDbRead() {
     try {
-      const { resources: results } = await this.cosmosDbClient
-        .database(this.databaseName)
-        .container(this.CONTAINER_NAME)
-        .items.readAll()
-        .fetchAll();
-      return results.length > 0;
+      const result = await this.getAdapter<HealthCheckDocument>().find(null);
+
+      const items = [];
+      for await (const doc of result) {
+        items.push(doc);
+      }
+      return items.length > 0;
     } catch (e) {
-      this.applicationContext.logger.error(MODULE_NAME, `${e.name}: ${e.message}`);
+      this.context.logger.error(MODULE_NAME, `${e.name}: ${e.message}`);
     }
     return false;
   }
 
   public async checkDbWrite() {
+    const healthCheckDocument: HealthCheckDocument = {
+      healtchCheckId: 'arbitrary-id',
+      documentType: 'HEALTH_CHECK',
+    };
     try {
-      const { resource } = await this.cosmosDbClient
-        .database(this.databaseName)
-        .container(this.CONTAINER_NAME)
-        .items.create({});
-      this.applicationContext.logger.debug(MODULE_NAME, `New item created ${resource.id}`);
-      return resource.id != undefined;
+      const resource = await this.getAdapter<HealthCheckDocument>().insertOne(healthCheckDocument);
+      this.context.logger.debug(MODULE_NAME, `New item created ${resource}`);
+      return !!resource;
```

Comment:
A couple of things here.
1. The response from `insertOne()` is not a resource, rather a string representation of the inserted resource's identifier.
2. I don't think we need to leave the logging in.